### PR TITLE
Update PS

### PR DESCRIPTION
**Pull request by VanMSFT**

### CODE REVIEW

**Comment by sptramer** [comment]
I'm still seeing the error from the previous pull request:

```
PS C:\Users\sttramer.REDMOND\src\azure-docs-powershell-samples> .\sql-database\transparent-data-encryption\setup-tde-byok-sqlmi.ps1
At C:\Users\sttramer.REDMOND\src\azure-docs-powershell-samples\sql-database\transparent-data-encryption\setup-tde-byok-sqlmi.ps1:36 char:146
+ ... yVault -ManagedInstanceName "MyManagedInstance" -ResourceGroup "MyRG"
+                                                                         ~
The string is missing the terminator: ".
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : TerminatorExpectedAtEndOfString
```

This error doesn't necessarily indicate a problem with the `MyRG` string, just that there's an unterminated string somewhere. It could be related to the interpretation of the provided "password" (BTW this is not secure, and should be changed to use `[guid]::NewGuid()`).

Here's some information about my running environment:

```
PS C:\Users\sttramer.REDMOND\src\azure-docs-powershell-samples> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.17763.134
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.134
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by VanMSFT** [comment]
I went through line by line and removed/re-added all hyphens and quotes. Let's see if this fixes it:

image

**Comment by sptramer** [comment]
That did fix the first round of errors, thank you. There are some other changes needed, so I'll add an additional part to the review.

**Comment by VanMSFT** [comment]
@sptramer - Thanks Stephen. This request was supposed to be a straight forward request to edit the property -Name to -ManagedInstanceName. The article that calls that script comes from here, and assumes the person has an AKV: https://docs.microsoft.com/en-us/azure/sql-database/scripts/transparent-data-encryption-byok-sql-managed-instance-powershell

However, I can see that it may be good to add a comment and command to add a RG if they do not have one. @MladjoA - Should we add the command to create the resource group first?

Also, isn't the command New-AzureRmResourceGroup instead of Create-AzureRmResourceGroup? 
https://docs.microsoft.com/en-us/powershell/module/azurerm.resources/new-azurermresourcegroup?view=azurermps-6.13.0

**Comment by VanMSFT** [comment]
@MladjoA - Hey Mladen, following up on this. What do you think about adding some lines to first create the resource group? I can add this to the script if you are in agreement.

**Comment by MladjoA** [comment]
@VanMSFT Van, sure, it won't hurt to have it. Please go ahead and add it. Thanks.

**Comment by VanMSFT** [comment]
Thanks @MladjoA! @sptramer - I implemented the recommendations. Please help merge. Thanks!

**Comment by sptramer** [comment]
Unfortunately, I'm still not able to get this script to work:

```
Set-AzureRmSqlManagedInstance : The term 'Set-AzureRmSqlManagedInstance' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again.
At C:\Users\sttramer.REDMOND\src\azure-docs-powershell-samples\sql-database\transparent-data-encryption\setup-tde-byok-sqlmi.ps1:12 char:13
+ $instance = Set-AzureRmSqlManagedInstance -ResourceGroupName "MyRG" - ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Set-AzureRmSqlManagedInstance:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

```
PS C:\Users\sttramer.REDMOND\src\azure-docs-powershell-samples> Get-Command Set-AzureRmSqlManagedInstance
Get-Command : The term 'Set-AzureRmSqlManagedInstance' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
At line:1 char:1
+ Get-Command Set-AzureRmSqlManagedInstance
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Set-AzureRmSqlManagedInstance:String) [Get-Command], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,Microsoft.PowerShell.Commands.GetCommandCommand
```

Checking the `AzureRM.SQL` documentation, this cmdlet is not available. Is it in a module that isn't included in the `AzureRM` rollup? Did it get renamed?

https://docs.microsoft.com/en-us/powershell/module/azurerm.sql/?view=azurermps-6.13.0

**Comment by MladjoA** [comment]
It’s in the preview branch, please see instructions within the article<https://docs.microsoft.com/en-us/azure/sql-database/scripts/transparent-data-encryption-byok-sql-managed-instance-powershell#launch-azure-cloud-shell>:

If you choose to install and use the PowerShell locally, this tutorial also requires preview version of AzureRM.Sql PowerShell package 4.11.6-preview. Run the following command to install it: Install-Module -Name AzureRM.Sql -RequiredVersion 4.11.6-preview -AllowPrerelease

Regards,
Mladen

**Comment by VanMSFT** [comment]
@MladjoA  @sptramer - I added the following lines to eliminate some confusion:

# Install the preview version of AzureRM.Sql PowerShell package 4.11.6-preview if you are running this PowerShell locally
Install-Module -Name AzureRM.Sql -RequiredVersion 4.11.6-preview -AllowPrerelease

Please let me know if anything else is needed.

File: sql-database/transparent-data-encryption/setup-tde-byok-sqlmi.ps1 [hunk 1]
```diff
@@ -2,12 +2,19 @@
 Connect-AzureRmAccount
 
 # If there are multiple subscriptions, choose the one where AKV is created: 
-Set-AzureRmContext -SubscriptionId <subscription ID>
+Set-AzureRmContext -SubscriptionId "subscription ID"
 
-# 1. Create and setup Azure Key Vault (skip if already done)
+# Install the preview version of AzureRM.Sql PowerShell package 4.11.6-preview if you are running this PowerShell locally
+Install-Module -Name AzureRM.Sql -RequiredVersion 4.11.6-preview -AllowPrerelease
```
[the remark this answers is not in the document]
Comment (VanMSFT) [reply sql-database/transparent-data-encryption/setup-tde-byok-sqlmi.ps1]: Commented:

image

File: sql-database/transparent-data-encryption/setup-tde-byok-sqlmi.ps1 [hunk 2]
```diff
@@ -28,15 +35,15 @@ Update-AzureRmKeyVaultNetworkRuleSet -VaultName "MyKeyVault" -DefaultAction Deny
 $key = Add-AzureKeyVaultKey -VaultName MyKeyVault -Name MyTDEKey -Destination Software -Size 2048
 
 # Alternatively, import existing key from .pfx file:
```
[the remark this answers is not in the document]
Comment (VanMSFT) [reply sql-database/transparent-data-encryption/setup-tde-byok-sqlmi.ps1]: Changed:

image

**Comment by VanMSFT** [comment]
@sptramer Changes have been pushed. Thanks!

**Pull request merged**